### PR TITLE
Wasm & Asm.js common RegisterSpace

### PR DESCRIPTION
**Pull request by Cellule**
Made a common class WAsmJs::RegisterSpace to be shared between asmjs and wasm 
This removed duplicated code and added a debug feature for wasm

---
This change is [Reviewable](https://reviewable.io/reviews/microsoft/chakracore/1330)

### CODE REVIEW

**Comment by Cellule** [comment]
@MikeHolman @commonlisp Review please

File: lib/Runtime/Language/WAsmjsUtils.cpp [hunk 1]
```diff
@@ -0,0 +1,51 @@
+//-------------------------------------------------------------------------------------------------------
+// Copyright (C) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
+// Portions of this file are copyright 2014 Mozilla Foundation, available under the Apache 2.0 license.
+//-------------------------------------------------------------------------------------------------------
+
+#pragma once
```
Comment (MikeHolman) [inline lib/Runtime/Language/WAsmjsUtils.cpp]: assuming this is typo?

File: lib/Runtime/Language/WasmAsmJsCommon.h [hunk 2]
```diff
@@ -0,0 +1,9 @@
+//-------------------------------------------------------------------------------------------------------
+// Copyright (C) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
+// Portions of this file are copyright 2014 Mozilla Foundation, available under the Apache 2.0 license.
+//-------------------------------------------------------------------------------------------------------
+
+#pragma once
+
+#include "WAsmJsUtils.h"
```
Comment (MikeHolman) [inline lib/Runtime/Language/WasmAsmJsCommon.h]: why not remove this file and put #include in Runtime.h?

Comment (Cellule) [inline lib/Runtime/Language/WasmAsmJsCommon.h]: I was planning to have multiple files (upcoming PR), but I think I will use only this one. Will make the change

**Comment by MikeHolman** [comment]
lgtm

**Pull request merged**